hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/HapiSpec.java

```diff
@@ -457,6 +485,132 @@ private void exec(List<HapiSpecOperation> ops) {
+        // One slot for the ScheduleCreate, one for each ScheduleSign,
+        // one for the GetScheduleInfo, and one for the GetTxnRecord
+        final var orderedOps = new HapiSpecOperation[1 + numSignTxns + 2];
+        final var num = NEXT_AUTO_SCHEDULE_NUM.getAndIncrement();
+        final var schedule = "autoScheduled" + num;
+        final var creation = "autoScheduleCreation" + num;
+        orderedOps[0] = scheduleCreate(schedule, txn)
+                .alsoSigningWithExplicit(signingKeys.subList(indices.get(0), indices.get(1)))
+                .savingExpectedScheduledTxnId()
+                .via(creation);
+        for (int i = 1, n = indices.size(); i < n - 1; i++) {
+            orderedOps[i] = scheduleSign(schedule)
+                    .alsoSigningWithExplicit(signingKeys.subList(indices.get(i), indices.get(i + 1)))
+                    // It's likely that some of the top-level transaction's signing keys will already
+                    // be present (e.g. the default payer's signature), so we accommodate that here
+                    // by adding NO_NEW_VALID_SIGNATURES to the list of acceptable statuses
+                    .hasKnownStatusFrom(SUCCESS, NO_NEW_VALID_SIGNATURES);
+        }
+
+        final var expectedStatus =
+                (txn.getExpectedPrecheck() == OK) ? txn.getExpectedStatus() : txn.getExpectedPrecheck();
+        final var scheduleStateAssertion = getScheduleInfo(schedule);
+        // If the original transaction was supposed to fail with INVALID_SIGNATURE,
+        // then the schedule should not have been executed
+        if (expectedStatus != INVALID_SIGNATURE) {
+            scheduleStateAssertion.isExecuted();
+        } else {
+            scheduleStateAssertion.isNotExecuted();
+        }
+        orderedOps[orderedOps.length - 2] = scheduleStateAssertion;
+
+        if (expectedStatus != INVALID_SIGNATURE) {
+            final var recordAssertion = getTxnRecord(creation)
+                    .scheduledBy(schedule)
+                    .hasPriority(recordWith().status(expectedStatus));
+            orderedOps[orderedOps.length - 1] = recordAssertion;
+        } else {
+            // If the original transaction was supposed to fail with INVALID_SIGNATURE, there
+            // will be no scheduled transaction record to retrieve, so just use noOp()
+            orderedOps[orderedOps.length - 1] = noOp();
+        }
+        return blockingOrder(orderedOps);
+    }
+
+    private boolean isEmpty(final Key key) {
+        if (key.hasKeyList()) {
+            return key.getKeyList().getKeysCount() == 0
+                    || key.getKeyList().getKeysList().stream().allMatch(this::isEmpty);
+        } else if (key.hasThresholdKey()) {
+            return key.getThresholdKey().getKeys().getKeysCount() == 0
+                    || key.getThresholdKey().getKeys().getKeysList().stream().allMatch(this::isEmpty);
+        } else {
+            return false;
+        }
+    }
```

Comment:
Is this really the only place a method like this exists? Seems like it would be useful in other places too, but maybe not

Reply:
Not sure, will keep it in mind. 😁